### PR TITLE
Add parameter to TrajectoryFollower stop rotation when bearing is reached

### PR DESCRIPTION
**Pull request by iche033**
Signed-off-by: Ian Chen <ichen@osrfoundation.org>
# 🎉 New feature

## Summary

Added parameter `<zero_vel_on_bearing_reached>`. When set to true, it forces angular velocity to be zero when the target bearing is reached.  This is intended to remove any residue angular velocity of the link and prevent overshooting. 

## Test it

Test with trajectory_follower.sdf world. `box_02` now has this parameter set and uses high torque for faster rotation and should not overshoot

```
ign gazebo -v 4 ./src/ign-gazebo/examples/worlds/trajectory_follower.sdf
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/trajectory_follower/TrajectoryFollower.cc [hunk 1]
```diff
@@ -376,11 +387,31 @@ void TrajectoryFollower::PreUpdate(
   {
     forceWorld = (*comPose).Rot().RotateVector(
       ignition::math::Vector3d(this->dataPtr->forceToApply, 0, 0));
-  }
 
+    // force angular velocity to be zero when bearing is reached
+    if (this->dataPtr->forceZeroAngVel &&
+        math::equal(std::abs(bearing.Degree()), 0.0, 1.0))
```
Comment (caguero) [inline src/systems/trajectory_follower/TrajectoryFollower.cc:393]: Do we need the second expression `math::equal(std::abs(bearing.Degree()), 0.0, 1.0)`?

We're supposed to be within the acceptable range bearing. If we need higher bearing accuracy, I think we should change the parameter `bearingTolerance`.

Comment (iche033) [reply src/systems/trajectory_follower/TrajectoryFollower.cc]: The behavior I found without this check is that as soon as the vehicle moves forward 1 iteration, its bearing immediately becomes outside of `bearingTolerance`. So it constantly switched back and forth between rotation and moving forward. The problem was that removing the `AngularVelocityCmd` component is found to be very expensive and this locks up simulation. With this check, I still do notice a slight pause in simulation when the component is being removed but it only happens when occasionally when bearing needs to be corrected.

Comment (iche033) [reply src/systems/trajectory_follower/TrajectoryFollower.cc]: I tweaked the 2nd tol check to be a percentage of `bearingTolerance`. [aa8cf68](https://github.com/ignitionrobotics/ign-gazebo/pull/1349/commits/aa8cf68ec8894ec110fa0df9c551d6d9ba755d30)

**Review by caguero — APPROVED** [review]
Looks good to me.

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**